internal/databases/sqlserver/log_restore_handler.go

```diff
@@ -56,22 +56,43 @@ func HandleLogRestore(backupName string, untilTS string, dbnames []string, fromn
 	err = runParallel(func(i int) error {
 		dbname := dbnames[i]
 		fromname := fromnames[i]
+		var lastPiece bool
+		var LogsMetadata []BackupProperties
+
 		for _, logBackupName := range logs {
-			ok, err := doesLogBackupContainDB(folder, logBackupName, fromname)
+			baseURL := getLogBackupURL(logBackupName, fromname)
+			basePath := getLogBackupPath(logBackupName, fromname)
+			blobs, err := listBackupBlobs(folder.GetSubFolder(basePath))
+			if err != nil {
+				return err
+			}
+			urls := buildRestoreUrls(baseURL, blobs)
+			trn, err := ListBackupProperties(db, urls, logBackupName)
 			if err != nil {
 				return err
 			}
-			if !ok {
-				// some log backup may not contain particular database in case
-				// it was created or dropped between base backups
+			LogsMetadata = append(LogsMetadata, trn[0])
+		}
+
+		logBeginTS := time.Date(1900, 1, 1, 1, 1, 1, 1, time.UTC)
+
+		for _, bak := range LogsMetadata {
+			if bak.DatabaseName != fromname {
```

Comment:
Useless check. We collect properties only for log backups for `fromname` (line 63, 64)
There should not be properties for other databases.